public/javascript/main.js

```diff
@@ -279,9 +278,7 @@ $(function(){
 
   $('.js-sortable').sortable();
 
-  $('.js-navigation-menu').on('change', function(){
-    window.location.href = $(this).val();
-  });
+  $('.js-navigation-menu').on('change', deferredTracking);
```

Comment:
You've removed `window.location.href = $(this).val();` from this `change` handler, but not introduced anything to replace it inside `deferredTracking`.

`deferredTracking` probably needs to either work out that it's been called on a `<select>` rather than an `<a>`, or you need to make `deferredTracking` return the deferred promise, so you can attach your own completion callback up here.